src/visualizers/P5Visualizer.ts

```diff
@@ -156,14 +144,30 @@ export abstract class P5Visualizer
      */
     view(seq: SequenceInterface): void {
         this.seq = seq
+        if (!this._sketch) return
+        const element = this.within!
+        this.stop()
+        this.depart(element)
+        this.inhabit(element)
+        this.show()
     }
 
     /**
      * Display the visualizer within the element that it is inhabiting.
      * All it has to do is call draw, since p5 calls setup for us.
      */
     show(): void {
-        this._sketch?.draw()
+        if (this._canvas) this._sketch?.draw()
+        else {
+            // If the rendering context is not yet ready, start an interval
+            // that waits until the canvas is ready and shows when finished
+            const interval = setInterval(() => {
+                if (this._canvas) {
+                    clearInterval(interval)
+                    this._sketch?.draw()
+                }
+            }, 5)
```

Comment:
Is that 5 milliseconds? Seems quite short... How many times is this cycling in practice? I guess if you're observing just one cycle of waiting, then 5 is OK. But if it's cycling a few times, it's certainly OK to go up to at least 20. Also, should probably give this a symbolic name so that a future maintainer can see what the number "5" is doing. Thanks.